File: src/main/java/com/menmar/gestionTienda/service/EstablecimientoService.java
package com.menmar.gestionTienda.service;

import com.menmar.gestionTienda.model.establecimiento.*;

import java.util.List;

public interface EstablecimientoService {
    EstablecimientoResponse crear(EstablecimientoRequest request);
    List<EstablecimientoResponse> listar();
    EstablecimientoResponse buscarPorId(Long id);
    EstablecimientoResponse actualizar(Long id, EstablecimientoRequest request);
    void desactivar(Long id);
    void asignarEmpleado(Long establecimientoId, Long usuarioId);
    void desasignarEmpleado(Long establecimientoId, Long usuarioId);

    // Precios personalizados
    List<PrecioEstablecimientoResponse> listarPreciosCalzado(Long establecimientoId);
    List<PrecioEstablecimientoResponse> listarPreciosCostura(Long establecimientoId);
    List<PrecioEstablecimientoResponse> listarPreciosLlave(Long establecimientoId);
    PrecioEstablecimientoResponse upsertPrecioCalzado(Long establecimientoId, PrecioEstablecimientoRequest request);
    PrecioEstablecimientoResponse upsertPrecioCostura(Long establecimientoId, PrecioEstablecimientoRequest request);
    PrecioEstablecimientoResponse upsertPrecioLlave(Long establecimientoId, PrecioEstablecimientoRequest request);
    void eliminarPrecioCalzado(Long establecimientoId, Long tipoReparacionId);
    void eliminarPrecioCostura(Long establecimientoId, Long tipoCosturaId);
    void eliminarPrecioLlave(Long establecimientoId, Long tipoLlaveId);

    // Precios sugeridos para el frontend al crear ticket
    List<PreciosSugeridosResponse> preciosSugeridosCalzado(Long establecimientoId);
    List<PreciosSugeridosResponse> preciosSugeridosCostura(Long establecimientoId);
    List<PreciosSugeridosResponse> preciosSugeridosLlave(Long establecimientoId);
}

File: frontend/src/service/EstablecimientoService.ts
import api from './ApiService'
import type {
    EstablecimientoRequest,
    EstablecimientoResponse,
    PrecioEstablecimientoRequest,
    PrecioEstablecimientoResponse,
    PreciosSugeridosResponse
} from '@/types'

const EstablecimientoService = {
    async listar(): Promise<EstablecimientoResponse[]> {
        const res = await api.get<EstablecimientoResponse[]>('/establecimientos')
        return res.data
    },

    async obtener(id: number): Promise<EstablecimientoResponse> {
        const res = await api.get<EstablecimientoResponse>(`/establecimientos/${id}`)
        return res.data
    },

    async crear(req: EstablecimientoRequest): Promise<EstablecimientoResponse> {
        const res = await api.post<EstablecimientoResponse>('/establecimientos', req)
        return res.data
    },

    async actualizar(id: number, req: EstablecimientoRequest): Promise<EstablecimientoResponse> {
        const res = await api.put<EstablecimientoResponse>(`/establecimientos/${id}`, req)
        return res.data
    },

    async eliminar(id: number): Promise<void> {
        await api.delete(`/establecimientos/${id}`)
    },

    async asignarEmpleado(id: number, usuarioId: number): Promise<void> {
        await api.post(`/establecimientos/${id}/empleados/${usuarioId}`)
    },

    async quitarEmpleado(id: number, usuarioId: number): Promise<void> {
        await api.delete(`/establecimientos/${id}/empleados/${usuarioId}`)
    },

    // Precios personalizados
    async getPreciosCalzado(id: number): Promise<PrecioEstablecimientoResponse[]> {
        const res = await api.get<PrecioEstablecimientoResponse[]>(`/establecimientos/${id}/precios/calzado`)
        return res.data
    },
    async upsertPrecioCalzado(id: number, req: PrecioEstablecimientoRequest): Promise<PrecioEstablecimientoResponse> {
        const res = await api.put<PrecioEstablecimientoResponse>(`/establecimientos/${id}/precios/calzado`, req)
        return res.data
    },
    async deletePrecioCalzado(id: number, tipoReparacionId: number): Promise<void> {
        await api.delete(`/establecimientos/${id}/precios/calzado/${tipoReparacionId}`)
    },

    async getPreciosCostura(id: number): Promise<PrecioEstablecimientoResponse[]> {
        const res = await api.get<PrecioEstablecimientoResponse[]>(`/establecimientos/${id}/precios/costura`)
        return res.data
    },
    async upsertPrecioCostura(id: number, req: PrecioEstablecimientoRequest): Promise<PrecioEstablecimientoResponse> {
        const res = await api.put<PrecioEstablecimientoResponse>(`/establecimientos/${id}/precios/costura`, req)
        return res.data
    },
    async deletePrecioCostura(id: number, tipoCosturaId: number): Promise<void> {
        await api.delete(`/establecimientos/${id}/precios/costura/${tipoCosturaId}`)
    },

    async getPreciosLlaves(id: number): Promise<PrecioEstablecimientoResponse[]> {
        const res = await api.get<PrecioEstablecimientoResponse[]>(`/establecimientos/${id}/precios/llaves`)
        return res.data
    },
    async upsertPrecioLlave(id: number, req: PrecioEstablecimientoRequest): Promise<PrecioEstablecimientoResponse> {
        const res = await api.put<PrecioEstablecimientoResponse>(`/establecimientos/${id}/precios/llaves`, req)
        return res.data
    },
    async deletePrecioLlave(id: number, tipoLlaveId: number): Promise<void> {
        await api.delete(`/establecimientos/${id}/precios/llaves/${tipoLlaveId}`)
    },

    // Precios sugeridos
    async getPreciosSugeridosCalzado(id: number): Promise<PreciosSugeridosResponse[]> {
        const res = await api.get<PreciosSugeridosResponse[]>(`/establecimientos/${id}/precios-sugeridos/calzado`)
        return res.data
    },
    async getPreciosSugeridosCostura(id: number): Promise<PreciosSugeridosResponse[]> {
        const res = await api.get<PreciosSugeridosResponse[]>(`/establecimientos/${id}/precios-sugeridos/costura`)
        return res.data
    },
    async getPreciosSugeridosLlaves(id: number): Promise<PreciosSugeridosResponse[]> {
        const res = await api.get<PreciosSugeridosResponse[]>(`/establecimientos/${id}/precios-sugeridos/llaves`)
        return res.data
    }
}

export default EstablecimientoService
